### PR TITLE
User crud

### PR DESCRIPTION
**Pull request by Manish21-v**

### CODE REVIEW

File: Mongo_Assignment/connection_Config.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+const password = process.env.PASSWORD;
+
+function connect()
+{
+  //Getting the Database credentails from environment variable!!!
+  console.log(`username is == ${username}`);
+  console.log(`password is ${password}`);
+
+    //Establishing connection with mongoDb!!!!!!
+    mongoClient.connect(`mongodb://${username}:${password}@127.0.0.1:27017`,function(err, db){
```
Comment (sunil4sonawane) [inline Mongo_Assignment/connection_Config.js]: add mongo url from ENV

File: Mongo_Assignment/connection_Config.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    });
+}
+
+//Connecting to the required Database!!!!
+function connectToDb(dbName) {
+  return dbObj.db(dbName);
+}
+
+module.exports = {connect,connectToDb};
+
```
Comment (sunil4sonawane) [inline Mongo_Assignment/connection_Config.js:33]: remove extra new line

File: Mongo_Assignment/server.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const express = require('express');
+const myRoute = require('./user');
+const app = express();
+
+const port = process.env.NODE_ENV
+
+const mydbConnection = require('./mongo_Connection');
+app.use('/user',myRoute);
```
Comment (sunil4sonawane) [inline Mongo_Assignment/server.js]: myRoute is not a valid name. Give some related name

File: Mongo_Assignment/server.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+//Established the connection with the mongoDb server
+mydbConnection.connect(function(err)
```
Comment (sunil4sonawane) [inline Mongo_Assignment/server.js]: mydbconnection should be dbConnection

File: Mongo_Assignment/user.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  myDbConnection.getUser().collection('userdetail').insertOne(req.body, function(err, res) 
+  {    
+    if (err) 
+    { //if inserting data fails
+      console.log(err);
+    }
+    console.log("1 document inserted");
+  });
+  
+  res.json(`json object sended`);
```
Comment (sunil4sonawane) [inline Mongo_Assignment/user.js]: you should send {"msg" : "user added"}

res.json('json object sended') is not a valid statement

**Review by sunil4sonawane — CHANGES_REQUESTED** [review]
Please rectify code as per comments

File: Mongo_Assignment/connection_Config.js (hunk 1 above)
Comment (Manish21-v) [reply Mongo_Assignment/connection_Config.js]: Added mongo server and port no through env variable

File: Mongo_Assignment/connection_Config.js (hunk 2 above)
Comment (Manish21-v) [reply Mongo_Assignment/connection_Config.js:33]: Done

File: Mongo_Assignment/server.js (hunk 3 above)
Comment (Manish21-v) [reply Mongo_Assignment/server.js]: done changed it to userFileRef

File: Mongo_Assignment/server.js (hunk 4 above)
Comment (Manish21-v) [reply Mongo_Assignment/server.js]: done

File: Mongo_Assignment/user.js (hunk 5 above)
Comment (Manish21-v) [reply Mongo_Assignment/user.js]: done

File: Mongo_Assignment/user.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+const bodyParser = require('body-parser');
+var userDetails;
+var status;
+
+router.use(bodyParser.json());
+
+const myDbConnection = require('./mongo_Connection');
+
+//Router to get all user Details !!!!
+router.get('/getdetails',function(req,res)
```
Comment (sunil4sonawane) [inline Mongo_Assignment/user.js]: should be getUsers

File: Mongo_Assignment/connection_Config.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+      }
+    });
+}
+
+//Connecting to the required Database!!!!
+function connectToDb(dbName) {
+  return dbObj.db(dbName);
+}
+module.exports = {connect,connectToDb};
+
```
Comment (sunil4sonawane) [inline Mongo_Assignment/connection_Config.js:33]: remove on extra blank line

**Review by sunil4sonawane — CHANGES_REQUESTED** [review]
Overall code is excellent. Please resolve 2 comments mentioned

**Review by sunil4sonawane — APPROVED** [review]

File: Mongo_Assignment/connection_Config.js (hunk 7 above)
Comment (Manish21-v) [reply Mongo_Assignment/connection_Config.js:33]: yes removed

**Pull request closed without merging**